ESP8266WirelessPrintAsync/ESP8266WirelessPrintAsync.ino

```diff
@@ -825,59 +851,55 @@ void ReceiveResponses() {
     char ch = (char)Serial.read();
     serialResponse += ch;
     if (ch == '\n') {
+      bool incompleteResponse = false;
+      String responseDetail = "";
+
       if (serialResponse.startsWith("ok", lineStartPos)) {
-        telnetSend("< " + serialResponse + "\r\n  " + millis() + "\r\n  free heap RAM: " + ESP.getFreeHeap() + "\r\n");
         if (lastCommandSent.startsWith(TEMP_COMMAND))
           parseTemperatures(serialResponse);
         else if (fwAutoreportTempCap && lastCommandSent.startsWith(AUTOTEMP_COMMAND))
           autoreportTempEnabled = (lastCommandSent[6] != '0');
-        GotValidResponse();   // Warning, this will empty 'serialResponse'
-        commandAcknowledged();
-      }
-      else if (parseTemperatures(serialResponse)) {
-        telnetSend("AutoReportTemps parsed");
-        restartSerialTimeout();   // When firmware doesn't have 'BUSY_WHILE_HEATING' temperature sent during heating may be used to prevent timeout
-        GotValidResponse();       // Warning, this will empty 'serialResponse'
-      }
-      else if (parsePosition(serialResponse)) {
-        telnetSend("MPosition parsed");
-        restartSerialTimeout();   // Some firmware doesn't send busy while homing but just position. It can be used to prevent timeout
-        GotValidResponse();       // Warning, this will empty 'serialResponse'
-      }
-      else if (serialResponse.startsWith("echo:busy")) {
-        telnetSend("Printer is busy, giving it more time");
-        restartSerialTimeout();
-        GotValidResponse();   // Warning, this will empty 'serialResponse'
-      }
-      else if (serialResponse.startsWith("echo: cold extrusion prevented")) {
-        telnetSend("Printer is cold, can't move");
-        // To do: Pause sending gcode, or do something similar
-        GotValidResponse();   // Warning, this will empty 'serialResponse'
-      }
-      else if (serialResponse.startsWith("Error:")) {
-        telnetSend("Error received: ");
-        telnetSend("<" + String(serialResponse) );
-        cancelPrint = true;
-        GotValidResponse();   // Warning, this will empty 'serialResponse'
+
+        unsigned int cmdLen = commandQueue.popAcknowledge().length();     // Go on with next command
+        printerUsedBuffer = max(printerUsedBuffer - cmdLen, 0u);
       }
       else {
-        lineStartPos = serialResponse.length();
-        telnetSend("Unhandled line received:");
-        telnetSend("<" + String(serialResponse) );
+        restartSerialTimeout();     // Something has been received so clear timeout (printer and communication are live
+
+        if (parseTemperatures(serialResponse))
+          responseDetail = "autotemp";
+        else if (parsePosition(serialResponse))
+          responseDetail = "position";
+        else if (serialResponse.startsWith("echo:busy"))
```

Comment:
No space after the ":" here? Compare to the next line

Reply:
in marlin message is "echo:busy: xxxxxx". only space after busy ':'
OT what do you do to send these post? with code reference? help...

Reply:
Both line are correct, checked Marlin 2, cold has space busy not